src/sqlfluff/rules/L050.py

```diff
@@ -101,8 +101,10 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
             # Non-whitespace segment.
             (context.segment.name not in whitespace_set)
             # We want first Non-whitespace segment so
-            # all preceding segments must be whitespace.
+            # all preceding segments must be whitespace
+            # and at least one is not meta.
             and all(segment.name in whitespace_set for segment in context.raw_stack)
+            and not all(segment.is_meta for segment in context.raw_stack)
```

Comment:
Can you add a test for this change?

Reply:
So for this one I discovered it needed adding as part of this PR. The issue was the bracket would have a meta character before it at the start of the file. and L050 would try and delete it, but then it would come back (and so on...).

So I guess the unit test for this change is the unit tests i've added for this rule? Haven't found another place where it comes up. I could add the same test query to L050.yml if you want though?

Reply:
This sounds like a case where you need a test that runs multiple rules. Tests like this are structured a bit differently, and they can be added to a subdirectory of `test/fixtures/linter/autofix`. Can you do that? That sounds better than simply adding the **same** test to L050.yml, because it sounds like it's the **interaction** of the rules that needs testing.

Reply:
@barrywhart Actually this issue was caused by the list comprehension in this rule

originally it was:
```python
[segment for segment in context.segment.segments if segment.name not in bracket_set],
```
essentially strips out the brackets but not the metas between the brackets and the inner statement (thats were the bad interaction with L050 described above was coming from).

I since changed that list comprehension to:
```python
[segment for segment in context.segment.segments if segment.name not in bracket_set and not segment.is_meta]
```
i.e. also exclude the metas, and just extract the wrapped statement.

so this interaction doesn't actually exist between the rules anymore so not sure adding it as a test would really validate anything?

Reply:
@barrywhart had a look at `test/fixtures/autofix` and for this case it would literally just be a repeat of the existing unit test, since L050 is (rightly) not even triggered anymore:
![image](https://user-images.githubusercontent.com/80432516/142512304-04099117-5cee-4279-beac-c572862db535.png)

I think it still makes sense to keep this adjustment in as it fixed the potential issue before my list comprehension fix above, but alternatively can just remove the change to this file as it technically wouldn't break any unit tests

Reply:
Now that I know the full context, it's fine to leave it as is. Testing decisions often involve nuance, and things evolve as you work on them. 😅 

As I often say, if something like this comes up again, we can always add a test then. It's so hard to guess how likely that is... 1%? 50% 🤷